examples/phi3/phi3_vision.py

```diff
@@ -0,0 +1,378 @@
+        external_data_name=pass_config.get("external_data_name", None),
+        size_threshold=pass_config.get("size_threshold", 1024),
+        convert_attribute=pass_config.get("convert_attribute", False),
+    )
+
+
+def main(raw_args=None):
+    args = get_args(raw_args)
+
+    output_dir = Path(args.output_dir).resolve()
+    output_dir.mkdir(parents=True, exist_ok=True)
+    cache_dir = Path(args.cache_dir).resolve()
+    cache_dir.mkdir(parents=True, exist_ok=True)
+
+    if args.inference and args.optimized_model_path:
+        generate(args.optimized_model_path)
+        return
+
+    input_model_path = output_dir / "phi3-vision-128k-instruct" / "pytorch"
+    if not is_model_ready(input_model_path):
+        print(f"Model not found from {input_model_path}, preparing the model...")
+        # prepare the input model
+        run_subprocess(
+            [
+                "bash",
+                "vision/scripts/prepare_phi3_vision_for_olive.sh",
+                str(output_dir),
+            ],
+            env=os.environ,
+        )
+    # if device is gpu, rewrite the genai_config.json
+    genai_config_path = output_dir / "genai_config.json"
+    with genai_config_path.open("r") as f:
+        genai_config = json.load(f)
+    if args.target == "cuda":
+        genai_config["model"]["decoder"]["session_options"]["provider_options"] = [{"cuda": {}}]
+    else:
+        genai_config["model"]["decoder"]["session_options"]["provider_options"] = []
+    with genai_config_path.open("w") as f:
+        json.dump(genai_config, f, indent=4)
+
+    build_text_embedding(args, input_model_path)
+
+    # Generate Olive configuration file for specific target
+    print("\nGenerating Olive configuration file...")
+    vision_config = generate_vision_config(args, input_model_path)
+    vision_output = olive_run(vision_config)
+    output_node = next(iter(vision_output.values())).get_top_ranked_nodes(1)[0]
+    model_path = output_node.model_config["config"]["model_path"]
+    pass_config = output_node.pass_run_config
+    resave_onnx_model(model_path, output_dir / "phi-3-v-128k-instruct-vision.onnx", pass_config)
+
+    text_config = generate_text_config(args, input_model_path)
+    try:
+        text_output = olive_run(text_config)
+        output_node = next(iter(text_output.values())).get_top_ranked_nodes(1)[0]
+        model_path = output_node.model_config["config"]["model_path"]
+        pass_config = output_node.pass_run_config
+        resave_onnx_model(model_path, output_dir / "phi-3-v-128k-instruct-text.onnx", pass_config)
+    except Exception:
+        config_link = "https://huggingface.co/microsoft/Phi-3-vision-128k-instruct-onnx-cpu/tree/main/cpu-int4-rtn-block-32-acc-level-4."
+        print(
+            "Ignore the error during Olive run.\nThis script will copy the "
+            "genai_config.json and processor_config.json from ",
+            config_link,
+        )
+        # even if the Olive run fails, partially generated model
```

Comment:
Do you know what the failure could be? Does it fail during quantization?

Reply:
Failure in model builder. They had a fix in main branch already. After the new version of genai is released, this failure will be resolved.

Reply:
https://github.com/microsoft/onnxruntime-genai/blob/cb313b49adec51f8679701df7d2451c9d21374a2/src/python/py/models/builder.py#L267C1-L269C129